src/intobj.h

```diff
@@ -109,14 +109,13 @@ EXPORT_INLINE Int ARE_INTOBJS(Obj o1, Obj o2)
 /* Note that the C standard does not define what >> does here if the
  * value is negative. So we have to be careful if the C compiler
  * chooses to do a logical right shift. */
+GAP_STATIC_ASSERT((-1) >> 1 == -1, "right shifts are not arithmetic");
+GAP_STATIC_ASSERT((-2) >> 1 == -1, "right shifts are not arithmetic");
```

Comment:
Why do you need the second test? I think the first should already distinguish arithmetic from logical shift.

Reply:
It's there for paranoia: in practice, >> could also do something else than arithmetic or logical shift, which then by chance might work for -1 but not -2... Of course we can never test all cases, but I figured, it wouldn't hurt to test one more...

Reply:
If you apply this logic here, shouldn't we make sure the left shifts also work? Or is there no reason to believe that, say, it is a circular shift?

Reply:
The C standard guarantees how left shifts work, and how right shifts for non-negative numbers work. Only right shifts of negatives are implementation specific. Presumably this is to enable efficient operations on both systems with twos complement and ones complement, or with a sign bit..

Reply:
Okay :)